usr/libexec/helper-scripts/account_state.sh

```diff
@@ -0,0 +1,213 @@
+  as_root
+  local user pass
+  user="${1:-}"
+  is_user "${user}"
+  if has getent; then
+    pass="$(getent shadow -- "${user}")"
+  else
+    pass="$(grep -e "^${user}:" -- /etc/shadow)"
+  fi
+  pass="${pass#"${user}:"*}"
+  pass="${pass%%":"*}"
+  printf '%s' "${pass}"
+}
+
+## Description: Get password without its state.
+## Output: Password trimmed from symbols.
+## Usage: get_clean_pass USER SYMBOL
+## Example: get_clean_pass user '\!\*'
+get_clean_pass(){
+  log info "${FUNCNAME[0]} $*"
+  local user pass symbol
+  user="${1:-}"
+  is_user "${user}"
+  symbol="${2:-'\*\!'}"
+  pass="$(get_pass "${user}" | tr -d "[${symbol}]")"
+  printf '%s\n' "${pass}"
+}
+
+## Description: Check if password is empty.
+## Output: None
+## Return: 0 when empty, 1 otherwise.
+## Usage: is_pass_empty USER
+## Example: is_pass_empty user
+is_pass_empty(){
+  log info "${FUNCNAME[0]} $*"
+  local user trim_pass
+  user="${1:-}"
+  is_user "${user}"
+  trim_pass="$(get_clean_pass "${user}")"
+  if test -z "${trim_pass}"; then
+    return 0
+  fi
+  return 1
+}
+
+## Description: Check if password is locked.
+## Output: None
+## Return: 0 when locked, 1 otherwise.
+## Usage: is_pass_locked USER
+## Example: is_pass_locked user
+is_pass_locked(){
+  log info "${FUNCNAME[0]} $*"
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if get_pass "${user}" | grep -qF '!'; then
+    return 0
+  fi
+  return 1
+}
+
+## Description: Check if password is disabled.
+## Output: None
+## Return: 0 when disabled, 1 otherwise.
+## Usage: is_pass_disabled USER
+## Example: is_pass_disabled user
+is_pass_disabled(){
+  log info "${FUNCNAME[0]} $*"
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if get_pass "${user}" | grep -qF '*'; then
+    return 0
+  fi
+  return 1
+}
+
+## Description: Lock a password.
+## Output: None
+## Return: passwd
+## Usage: lock_pass USER
+## Example: lock_pass user
+lock_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has passwd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if is_pass_locked "${user}"; then
+    return 0
+  fi
+  passwd -ql -- "${user}"
+}
+
+## Description: Unlock a password.
+## Output: None
+## Return: chpasswd
+## Usage: lock_pass USER
+## Example: lock_pass user
+unlock_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if ! is_pass_locked "${user}"; then
+    return 0
+  fi
+  ## The tools "passwd" and "usermod" can't unlock when password is empty.
+  pass="$(get_clean_pass "${user}" '\!')"
+  printf '%s' "${user}:${pass}" | chpasswd -c NONE
+}
+
+## Description: Disable a password.
+## Output: None
+## Return: chpasswd
+## Usage: disable_pass USER
+## Example: disable_pass user
+disable_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if is_pass_disabled "${user}"; then
+    return 0
+  fi
+  pass="$(get_pass "${user}")"
+  printf '%s' "${user}:*${pass}" | chpasswd -c NONE
```

Comment:
Would like a similar safety comment here if that's OK. Something that referred back to the first comment would brevity would be fine.